server/src/main/java/org/opensearch/search/pipeline/PipelineWithMetrics.java

```diff
@@ -139,8 +140,9 @@ private static <T extends Processor> List<T> readProcessors(
                 }
                 Map<String, Object> config = (Map<String, Object>) entry.getValue();
                 String tag = ConfigurationUtils.readOptionalStringProperty(null, null, config, TAG_KEY);
+                Boolean ignoreFailure = ConfigurationUtils.readBooleanProperty(null, null, config, IGNORE_FAILURE_KEY, false);
```

Comment:
This should be a lower-cased `boolean`. The value returned from `readBooleanProperty` is guaranteed to never be `null`. (While you're here, please update it to return `boolean` -- the fact that it returns `Boolean` looks like an old mistake.)